File: backend/src/schemas/invite.py
import uuid
from datetime import datetime

from pydantic import BaseModel, EmailStr, Field

from schemas._common import Page
from schemas.role import RoleRead


class InviteCreate(BaseModel):
    email: EmailStr
    role_id: uuid.UUID
    expires_in_hours: int = Field(default=168, ge=1, le=24 * 30)


class InvitedBySummary(BaseModel):
    id: uuid.UUID
    name: str


class InviteRead(BaseModel):
    id: uuid.UUID
    email: str
    role: RoleRead
    invited_by: InvitedBySummary | None = None
    token: str
    accepted_at: datetime | None = None
    expires_at: datetime
    created_at: datetime


InvitePage = Page[InviteRead]

File: frontend/src/lib/schemas/invite.ts
import { z } from "zod";

import { roleReadSchema } from "@/lib/schemas/role";

export const inviteCreateSchema = z.object({
  email: z.email().max(255),
  role_id: z.string().min(1),
  expires_in_hours: z.number().int().min(1).max(24 * 30).optional(),
});

export const invitedBySummarySchema = z.object({
  id: z.string(),
  name: z.string(),
});

export const inviteReadSchema = z.object({
  id: z.string(),
  email: z.string(),
  role: roleReadSchema,
  invited_by: invitedBySummarySchema.nullable().optional(),
  token: z.string(),
  accepted_at: z.string().nullable().optional(),
  expires_at: z.string(),
  created_at: z.string(),
});

export const invitePageSchema = z.object({
  items: z.array(inviteReadSchema),
  total: z.number(),
  page: z.number(),
  page_size: z.number(),
  has_more: z.boolean(),
});

export type InviteCreate = z.infer<typeof inviteCreateSchema>;
export type InvitedBySummary = z.infer<typeof invitedBySummarySchema>;
export type InviteRead = z.infer<typeof inviteReadSchema>;
export type InvitePage = z.infer<typeof invitePageSchema>;
